skimage/segmentation/_join.py

```diff
@@ -121,24 +121,21 @@ def relabel_sequential(label_field, offset=1):
         raise ValueError("Offset must be strictly positive.")
     if np.min(label_field) < 0:
         raise ValueError("Cannot relabel array that contains negative values.")
-    m = label_field.max()
+    max_label = label_field.max()
     if not np.issubdtype(label_field.dtype, np.integer):
-        new_type = np.min_scalar_type(int(m))
+        new_type = np.min_scalar_type(int(max_label))
         label_field = label_field.astype(new_type)
-        m = m.astype(new_type)  # Ensures m is an integer
+        max_label = max_label.astype(new_type)  # Ensures max_label is an integer
     labels = np.unique(label_field)
     labels0 = labels[labels != 0]
-    required_type = np.min_scalar_type(offset + len(labels0))
+    new_max_label = offset - 1 + len(labels0)
+    new_labels0 = np.arange(offset, int(new_max_label + 1))
```

Comment:
No need for explicitely casting to int
```suggestion
    new_labels0 = np.arange(offset, new_max_label + 1)
```

Reply:
Hi @rfezzani, you are right about that. I was wrongly assuming that `int(v)` would not change the data type of `v` if it was "integer-based" (e.g. `np.uint16`).

I went ahead and changed the code some more, as seen in the latest commit.